### ISSUE TITLE
Proposal: Add the ability to create a Websocket on an existing socket.

### ISSUE THREAD

**Issue by mea405** [daltoniam/Starscream#539]
Add the ability to create a Websocket on an existing socket.

**Comment by AFcgi** [comment]
Did somebody have experiences with this topic? Any solutions for this problem?

**Comment by texuf** [comment]
What is the problem exactly? You have an open socket that hasn't performed an HTTP-compatible handshake and you want to initiate the handshake? Maybe some examples or source code would help.

**Comment by AFcgi** [comment]
I did the following steps:

1. I instantiate a `WebSocket` object with a custom header (`URL`, `URLRequest`) and set the `delegate`
2. I connect the socket via `socket.connect()`
3. After that the socket is connected
4. I did some actions with the socket. If the actions are finished, I disconnect the socket.
5. After some seconds I try to change the `URL` from the instantiated `WebSocket` object. I change the `URL` from the `URLRequest` and set the changed `URLRequest` via `socket.request = URLRequest`.
6. After this it's impossible to `connect` or `reconnect` the socket with the new `URL`.

I tried also to reinitialize my `WebSocket` object. But that doesn't work either.
